agent/api/task/task.go

```diff
@@ -1699,23 +1723,43 @@ func (task *Task) PopulateSecretsAsEnv(container *apicontainer.Container) *apier
 
 	envVars := make(map[string]string)
 
+	LogDriverName := hostConfig.LogConfig.Type
+	LogDriverTokenSecretValue := ""
+
 	for _, secret := range container.Secrets {
-		if secret.Provider == apicontainer.SecretProviderSSM && secret.Type == apicontainer.SecretTypeEnv {
-			k := secret.GetSecretResourceCacheKey()
+		k := secret.GetSecretResourceCacheKey()
+		if secret.Provider == apicontainer.SecretProviderSSM {
 			if secretValue, ok := ssmRes.GetCachedSecretValue(k); ok {
-				envVars[secret.Name] = secretValue
+				if secret.Type == apicontainer.SecretTypeEnv {
+					envVars[secret.Name] = secretValue
+				}
+				if secret.Target == apicontainer.SecretTargetLogDriver {
+					LogDriverTokenSecretValue = secretValue
+				}
 			}
 		}
-
-		if secret.Provider == apicontainer.SecretProviderASM && secret.Type == apicontainer.SecretTypeEnv {
-			k := secret.GetSecretResourceCacheKey()
+		if secret.Provider == apicontainer.SecretProviderASM {
 			if secretValue, ok := asmRes.GetCachedSecretValue(k); ok {
-				envVars[secret.Name] = secretValue
+				if secret.Type == apicontainer.SecretTypeEnv {
+					envVars[secret.Name] = secretValue
+				}
+				if secret.Target == apicontainer.SecretTargetLogDriver {
+					LogDriverTokenSecretValue = secretValue
+				}
 			}
 		}
 	}
 
-	container.MergeEnvironmentVariables(envVars)
+	if len(envVars) != 0 {
+		container.MergeEnvironmentVariables(envVars)
+	}
+
+	//Check if the "splunk-token" has already been initialized and stored in the LogCofig map
```

Comment:
We should not include logic specific to Splunk or keys like "splunk-token".  If we need logic like this, it should be handled in the ECS backend so we do not need to keep updating the agent.

Reply:
I agree - this was discussed during design phase, to not cater implementation to one specific log driver.

Reply:
Figured out a way. Will discuss and update this pr shortly.